query/src/test/java/com/connexta/replication/query/QueryManagerTest.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (c) Connexta
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+
+package com.connexta.replication.query;
+
+import static org.junit.Assert.*;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.connexta.replication.api.data.Site;
+import com.connexta.replication.api.data.SiteType;
+import com.connexta.replication.api.queue.SiteQueue;
+import java.util.Map;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Stream;
+import org.hamcrest.Matchers;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class QueryManagerTest {
+
+  private static final long PERIOD = 60L;
+
+  private static final String SITE_ID = "siteid";
+
+  @Mock QueryServiceTools queryServiceTools;
+
+  @Mock ScheduledExecutorService executor;
+
+  @Mock private Site site;
+
+  @Mock private QueryService queryService;
+
+  private Stream<String> sites = Stream.empty();
+
+  private QueryManager queryManager;
+
+  @Before
+  public void setUp() throws Exception {
+    queryManager = new QueryManager(sites, queryServiceTools, executor);
+    when(queryServiceTools.getPeriod()).thenReturn(PERIOD);
+    when(site.getId()).thenReturn(SITE_ID);
+    when(site.getType()).thenReturn(SiteType.DDF);
+  }
+
```

Comment:
✏️ We should test cases where no site ids are defined (what we have here) and some when site ids are defined.

Reply:
✏️ We might want to add a test case that will:
- add a new site to manage
- remove an existing site to manage
- add a new site that would be filtered

Just to make sure that we do handle the combination